### PR TITLE
Add check for worker tag propagation to improve performance

### PR DESCRIPTION
**Pull request by satvu**
### Issue describing the changes in this PR

Updates a `null` check both null / empty mapfield to avoid execution context sync/change when no worker tags are propagated.

### Pull request checklist

**IMPORTANT**: Currently, changes must be backported to the `in-proc` branch to be included in Core Tools and non-Flex deployments.

* [x] Backporting to the `in-proc` branch is not required
    * Otherwise: Link to backporting PR
* [x] My changes **do not** require documentation changes
    * [ ] Otherwise: Documentation issue linked to PR
* [ ] My changes **should not** be added to the release notes for the next release
    * [x] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
    * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] My changes **do not** require diagnostic events changes
    * Otherwise: I have added/updated all related diagnostic events and their documentation (Documentation issue linked to PR)
* [x] I have added all required tests (Unit tests, E2E tests)

### CODE REVIEW

**Review by kshyju — APPROVED** [review]
Do we need any test updates?

**Comment by satvu** [comment]
> Do we need any test updates?

We could add a test to make sure [some step] doesn't happen if there's an empty/null worker tag collection but I didn't think this really required that coverage/check.

File: src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs [hunk 1]
```diff
@@ -16,6 +16,7 @@
 using System.Threading.Tasks.Dataflow;
 using Google.Protobuf.Collections;
 using Google.Protobuf.WellKnownTypes;
+using Microsoft.Azure.AppService.Proxy.Common.Extensions;
```
Comment (RohitRanjanMS) [inline src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs]: This using `Microsoft.Azure.AppService.Proxy.Common.Extensions` looks unnecessary — MapField<string, string>already has a native .Count property via IDictionary<TKey, TValue>.

**Review by RohitRanjanMS — CHANGES_REQUESTED** [review]

**Review by RohitRanjanMS — APPROVED** [review]
